librz/core/canalysis.c

```diff
@@ -7262,6 +7262,22 @@ RZ_IPI char *rz_core_analysis_all_vars_display(RzCore *core, RzAnalysisFunction
 	return rz_strbuf_drain(sb);
 }
 
+RZ_API bool rz_analysis_add_io_registers_map(RzIO *io, RzAnalysis *analysis, ut64 file_offset) {
+	ut64 rom_size = analysis->arch_target->profile->rom_size;
+	ut64 rom_address = analysis->arch_target->profile->rom_address;
+	RzIOMap *map = rz_io_map_get(io, rom_address);
+	if (!map) {
+		return false;
+	}
+	if (file_offset) {
```

Comment:
i would say here, that `if (file_offset >= rom_address && file_offset < (rom_address + rom_size))` then throw that warning.